murmur128.go

```diff
@@ -26,30 +26,30 @@ type Hash128 interface {
 	Sum128() (uint64, uint64)
 }
 
-// digest128 represents a partial evaluation of a 128 bites hash.
-type digest128 struct {
+// Digest128 represents a partial evaluation of a 128 bites hash.
+type Digest128 struct {
 	digest
 	h1 uint64 // Unfinalized running hash part 1.
 	h2 uint64 // Unfinalized running hash part 2.
 }
 
 // New128 returns a 128-bit hasher
-func New128() Hash128 { return New128WithSeed(0) }
+func New128() *Digest128 { return New128WithSeed(0) }
 
 // New128WithSeed returns a 128-bit hasher set with explicit seed value
-func New128WithSeed(seed uint32) Hash128 {
-	d := new(digest128)
+func New128WithSeed(seed uint32) *Digest128 {
+	d := new(Digest128)
 	d.seed = seed
 	d.bmixer = d
```

Comment:
I think this conversion to the `bmixer` interface type will always force `d` to be heap allocated because escape analysis doesn't know that calls to the interface methods won't retain the pointer.

```go
type bmixer interface {
    bmix(p []byte) (tail []byte)
    Size() (n int)
    reset()
}
```

Here's a strategy we could use:

* remove the `bmixer` type

* change `(*digest).Write(b []byte)` to 
```go
func (*digest) write(b []byte, size int, bmix func([]byte) []byte) (int, error)
```
and have each `Digest*` type implement `io.Writer` with as:
```go
func (d *Digest128) Write(b []byte) (int, error) { return d.write(b, d.Size(), d.bmix) }
```

* change `(*digest).Reset` to `(*digest).reset` and have each `Digest*` type provide the exported `Reset` method:
```go
func (d *Digest128) Reset() {
  d.reset() // generic reset of the embedded `digest` field
  ...       // type-specific reset
}
```